frontend/simple/views/containers/contributions/IncomeForm.vue

```diff
@@ -0,0 +1,245 @@
+    order: -1; // display on the top for better understanding on small screens
+    padding: $gi-spacer $gi-spacer-sm;
+  }
+
+  @include tablet {
+    justify-content: center;
+    margin-top: -$gi-spacer-lg;
+  }
+}
+
+</style>
+<script>
+import { validationMixin } from 'vuelidate'
+import { requiredIf, required, maxValue } from 'vuelidate/lib/validators'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Modal from '../../components/Modal/ModalBasic.vue'
+import TextWho from '../../components/TextWho.vue'
+import { debounce } from '../../../utils/giLodash.js'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+
+export default {
+  name: 'IncomeForm',
+  mixins: [ validationMixin ],
+  components: {
+    Modal,
+    InputAmount,
+    TextWho,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  props: {
+    isEditing: Boolean
+  },
+  data () {
+    return {
+      ephemeral: {
+        formVerified: false
+      },
+      form: {
+        option: null,
+        income: null,
+        pledge: null
+      },
+      // -- Hardcoded Data just for layout purposes:
+      fakeStore: {
+        mincome: 500,
+        mincomeFormatted: '$500'
+      }
+    }
+  },
+  computed: {
+    userPledgeAmount () {
+      return this.hasMinIncome & !!this.form.pledge ? Number(this.form.pledge) : null
+    },
+    userIncomeAmount () {
+      return this.hasLessIncome & !!this.form.income ? Number(this.form.income) : null
+    },
+    maxIncome () {
+      return this.fakeStore.mincome - 1
+    },
+    infoRequired () {
+      return this.L('This information is required.')
+    },
+    hasLessIncome () {
+      return this.$v.form.option.$model === 'no'
+    },
+    hasMinIncome () {
+      return this.$v.form.option.$model === 'yes'
+    },
+    hasLessIncomeError () {
+      if (!this.ephemeral.formVerified && this.$v.form.income.maxValue) {
+        return null
+      }
+
+      if (!this.$v.form.income.requiredIf) {
+        return this.infoRequired
+      } else if (!this.$v.form.income.maxValue) {
+        return this.L('If your income is the same or higher than the group\'s mincome change your answer to "Yes, I do"')
+      }
+    },
+    hasMinIncomeError () {
+      return this.ephemeral.formVerified && this.$v.form.pledge.$invalid ? this.infoRequired : null
+    },
+    saveButtonText () {
+      const verb = this.isFirstTime ? 'Add' : 'Save'
+
+      if (!this.$v.form.option.$model) { return this.L('{verb} details', { verb }) }
+      if (this.hasMinIncome) { return this.L('{verb} pledged details', { verb }) }
+      if (this.hasLessIncome) { return this.L('{verb} income details', { verb }) }
+    }
+  },
+  methods: {
+    verifyInputIncome: debounce(function (e) {
+      this.form.income = e.target.value
+      this.$v.form.income.$touch()
+    }, 0),
+    verifyInputPledge: debounce(function (e) {
+      this.form.pledge = e.target.value
+      this.$v.form.pledge.$touch()
+    }, 0),
```

Comment:
Why are these both being debounced with a zero delay?

Reply:
ups, a note was missing here: `REVIEW: Should we use debounce so the pie chart only react to input change after XXms or should it be immediately?`

After yesterday's call, we decided to remove it.